### PR TITLE
MM-47298 : Migrate /enterprise/bot_accounts e2e tests to TypeScript

### PR DESCRIPTION
**Pull request by Yordaniss**
#### Summary
Migrated e2e test from JavaScript to Typescript with type declaration.

#### Ticket Link

Fixes https://github.com/mattermost/mattermost-server/issues/21310

JIRA: https://mattermost.atlassian.net/browse/MM-47298

#### Screenshots

Bildschirmfoto 2022-10-13 um 20 42 33

#### Release Note

NONE

### CODE REVIEW

File: e2e/cypress/tests/integration/enterprise/bot_accounts/managing_bot_accounts_spec.ts [hunk 1]
```diff
@@ -10,26 +10,33 @@
...
+            const newConfig = {
+                ...origConfig,
+                ServiceSettings: {
+                    ...origConfig.ServiceSettings,
+                    EnableBotAccountCreation: true,
+                },
+            };
+
+            cy.apiUpdateConfig(newConfig);
+        });
```
Comment (saturninoabril) [inline e2e/cypress/tests/integration/enterprise/bot_accounts/managing_bot_accounts_spec.ts:37]: Please update to latest master for it has [`DeepPartial` applied to apiUpdateConfig](https://github.com/mattermost/mattermost-webapp/blob/master/e2e/cypress/tests/support/api/system.d.ts#L100). Then revert the workaround wrapping with `cy.apiGetConfig()`.

File: e2e/cypress/tests/support/api/bots.ts [hunk 2]
```diff
@@ -1,14 +1,16 @@
...
+
 import {getRandomId} from '../../utils';
 
 // *****************************************************************************
 // Bots
 // https://api.mattermost.com/#tag/bots
 // *****************************************************************************
 
-Cypress.Commands.add('apiCreateBot', ({prefix, bot = createBotPatch(prefix)} = {}) => {
+function apiCreateBot({prefix, bot = createBotPatch(prefix)}): Cypress.Chainable<{bot: Bot & {fullDisplayName: string}}> {
```
Comment (saturninoabril) [inline e2e/cypress/tests/support/api/bots.ts:13]: We're no longer using types from `mattermost-redux`. Instead, you may use directly from the Cypress namespace where types are imported in one place ([here](https://github.com/mattermost/mattermost-webapp/blob/master/e2e/cypress/tests/support/index.d.ts#L7-L35)).
```suggestion
function apiCreateBot({prefix, bot = createBotPatch(prefix)}): Cypress.Chainable<{bot: Cypress.Bot & {fullDisplayName: string}}> 
```

File: e2e/cypress/tests/support/api/bots.ts [hunk 3]
```diff
@@ -24,9 +26,11 @@ Cypress.Commands.add('apiCreateBot', ({prefix, bot = createBotPatch(prefix)} = {
             },
         });
     });
-});
+}
 
-Cypress.Commands.add('apiGetBots', () => {
+Cypress.Commands.add('apiCreateBot', apiCreateBot);
+
+function apiGetBots(): Cypress.Chainable<{bots: Bot[]}> {
```
Comment (saturninoabril) [inline e2e/cypress/tests/support/api/bots.ts:33]: Same here, please use `Cypress.Bot`.

**Review by saturninoabril — CHANGES_REQUESTED** [review]
Thanks @Yordaniss! Please see comments.

**Comment by Yordaniss** [comment]
Hello @saturninoabril Thank you for feedback. 👍  I have updated the master branch and adjusted my PR.

**Review by saturninoabril — CHANGES_REQUESTED** [review]
Thanks @Yordaniss, sorry last couple comments and I'm all good, thanks!

**Comment by Yordaniss** [comment]
@saturninoabril Sure, no problem. I already pushed the changes.

**Review by furqanmlk — APPROVED** [review]
Thanks @Yordaniss 
LGTM !

**Comment by Yordaniss** [comment]
@furqanmlk Glad to help 😊

**Comment by lieut-data** [comment]
Heads up that as part of our efforts to move to a monorepo, we're closing out a number of older pull requests like this one to help streamline the effort.

If you'd like to preserve these changes -- even if you're not the original author! -- feel free to resubmit the pull request against the monorepo once it's ready. You can subscribe to [mattermost-server-issue-22420](https://www.github.com/mattermost/mattermost-server/issues/22420) for status updates on this effort.

**Pull request closed without merging**